pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -6587,6 +6664,179 @@ def _preprocess_data(model):
+            for t in model.s_T:
+                if (
+                    sum(
+                        model.v_F_Piped[l, treatment_site, t].value
+                        for l in model.s_L
+                        if (l, treatment_site) in model.s_LLA
+                    )
+                    + sum(
+                        model.v_F_Trucked[l, treatment_site, t].value
+                        for l in model.s_L
+                        if (l, treatment_site) in model.s_LLT
+                    )
+                    > model.p_sigma_Treatment[treatment_site, i[1]].value
+                ):
+                    # Add first use to a dictionary
+                    model.infrastructure_firstUse[treatment_site] = t
+                    # Get the lead time rounded up to the nearest full time period
+                    model.infrastructure_leadTime[treatment_site] = math.ceil(
+                        model.p_tau_TreatmentExpansionLeadTime[i].value
+                    )
+                    break
+
+    # Disposal - "vb_y_Disposal"
+    disposal_data = model.vb_y_Disposal._data
+    for i in disposal_data:
+        # Get site name from data
+        disposal_site = i[0]
+        # add values to output dictionary
+        if (
+            disposal_data[i].value >= 1 - binary_epsilon
+            and disposal_data[i].value <= 1 + binary_epsilon
+            and model.p_delta_Disposal[i[1]].value > 0  # selected capacity is nonzero
+        ):
+            # determine first time period that site is used
+            # First use is time period where more water is sent to disposal than initial capacity
+            for t in model.s_T:
+                if (
+                    model.v_F_DisposalDestination[disposal_site, t].value
+                    > model.p_sigma_Disposal[disposal_site].value
+                ):
+                    # Add first use to a dictionary
+                    model.infrastructure_firstUse[disposal_site] = t
+                    # Get the lead time rounded up to the nearest full time period
+                    model.infrastructure_leadTime[disposal_site] = math.ceil(
+                        model.p_tau_DisposalExpansionLeadTime[i].value
+                    )
+                    break
+
+    # Storage - "vb_y_Storage"
+    storage_data = model.vb_y_Storage._data
+    # Storage Site - iterate through vb_y variables
+    for i in storage_data:
+        # Get site name from data
+        storage_site = i[0]
+        # add values to output dictionary
+        if (
+            storage_data[i].value >= 1 - binary_epsilon
+            and storage_data[i].value <= 1 + binary_epsilon
+            and model.p_delta_Storage[i[1]].value > 0  # selected capacity is nonzero
+        ):
+            # determine first time period that site is used
+            # a new storage site is first used when the water level is first above the initial water capacity
```

Comment:
I think this comment is confusingly worded - I recommend the following rewording:
```suggestion
            # First use is when the water level exceeds the initial storage capacity
```